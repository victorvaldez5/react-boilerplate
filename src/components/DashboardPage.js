import React from 'react';

const DashboardPage = (props) => (
  <div>
    Dashboard page content
  </div>
);

export default DashboardPage;