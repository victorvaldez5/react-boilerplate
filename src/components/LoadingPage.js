import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <div className="logo">
      <p>Loading</p>
      <div className="green"></div>
      <div className="purple"></div>
      <div className="blue"></div>
    </div>
  </div>
);

export default LoadingPage;