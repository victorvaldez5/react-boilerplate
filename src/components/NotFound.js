import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    <h1>404! Page Not Found!</h1>
    <h3>Return to <Link to="/">Dashboard</Link></h3>
  </div>
);

export default NotFoundPage;