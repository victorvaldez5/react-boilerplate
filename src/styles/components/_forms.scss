.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
}

.form__error{
  background: fade-out($danger, .35);
  margin: 0 0 $m-size 0;
  text-align: center;
  padding: 0 $s-size;
  color: white;
  font-style: italic;
  border-radius: 25px;
  border: 1px solid $danger;
}