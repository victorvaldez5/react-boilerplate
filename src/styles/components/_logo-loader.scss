$color: black;
$blue: #5BBDFF;
$purple: #BF5BFF;
$orange: #FFBE5B;
$size: 100px;
$borderWidth: 4px;
$blueWidth: 35%;
$purpleHeight: 50%;
$orangeWidth: 65%;
$totalTime: 1.75s;

div.loader {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 100;

  p {
    font-family: 'Helvetica', 'Arial', sans-serif;
    margin: 1em 0 0 0;
    font-size: 16px;
  }
}
div.logo {
  width: $size;
  height: $size;
  box-sizing: border-box;
  position: relative;
  background-color: white;
  text-align: center;
    &::before,
    &::after {
      z-index: 1;
      box-sizing: border-box;
      content: '';
      position: absolute;
      border: $borderWidth solid transparent;
      width: 0;
      height: 0;
      animation-timing-function: linear;
    }

    &::before {
      top: 0;
      left: 0;
      border-top-color: black;
      border-right-color: black; 
      animation: border-before $totalTime infinite;
      animation-direction: alternate;
    }

    &::after {
      bottom: 0;
      right: 0;
      border-bottom-color: $blue;
      border-left-color: $blue;
      animation: border-after $totalTime infinite;
      animation-direction: alternate;
    }
    & > div {
      position: absolute;
      opacity: 0;
    }
    div.blue {
      border-right: $borderWidth solid $color;
      top: 0;
      bottom: 0;
      left: 0;
      width: 0;
      background-color: #5BBDFF;
      animation: blue $totalTime infinite;
      animation-direction: alternate;
    }
    div.purple {
      border-top: $borderWidth solid $color;
      right: 0;
      bottom: 0;
      left: 0;
      height: 0;
      background-color: #BF5BFF;
      animation: purple $totalTime infinite;
      animation-direction: alternate;
    }
    div.green {
      border-left: $borderWidth solid $color;
      top: 0;
      bottom: 0;
      right: 0;
      width: 0;
      background-color: #FFBE5B;
      animation: orange $totalTime infinite;
      animation-direction: alternate;
    }
}

@keyframes orange {
  0%, 75% {
    width: 0;
    opactiy: 0;
  }
  75.01% {
    opacity: 1;
  }
  90%,
  100% {
    opacity: 1;
    width: $orangeWidth
  }
}

@keyframes purple {
  0%, 65% {
    height: 0;
    opactiy: 0;
  }
  65.01% {
    opacity: 1;
  }
  80%,
  100% {
    opacity: 1;
    height: $purpleHeight
  }
}

@keyframes blue {
  0%, 50% {
    width: 0;
    opactiy: 0;
  }
  50.01% {
    opacity: 1;
  }
  65%,
  100% {
    opacity: 1;
    width: $blueWidth
  }
}

@keyframes border-before {
  0% {
    width: 0;
    height: 0;
    border-top-color: black;
    border-right-color: transparent;
  }
  12.49% {
    border-right-color: transparent;
  }
  12.5% {
    height: 0;
    width: 100%;
    border-top-color: black;
    border-right-color: black;
    
  }
  25%,
  100% {
    width: 100%;
    height: 100%;
    border-top-color: black;
    border-right-color: black;
  }
}

@keyframes border-after {
  0%,
  24.99% {
    width: 0;
    height: 0;
    border-bottom-color: transparent;
    border-left-color: transparent;
  }
  25% {
    width: 0;
    height: 0;
    border-left-color: transparent;
    border-bottom-color: black;
  }
  37.99% {
    border-left-color: transparent;
    border-bottom-color: black;
  }
  37.5% {
    height: 0;
    width:100%;
    border-bottom-color: black;
    border-left-color: black;
  }
  50%,
  100% {
    width: 100%;
    height: 100%;
    border-bottom-color: black;
    border-left-color: black;
  }
}
// div.blue {
//   border-right: $borderWidth solid $color;
//   top: 0;
//   bottom: 0;
//   left: 0;
//   width: $blueWidth;
//   background-color: #5BBDFF;
// }
// div.purple {
//   border-top: $borderWidth solid $color;
//   right: 0;
//   bottom: 0;
//   left: 0;
//   height: $purpleHeight;
//   background-color: #BF5BFF;
// }
// div.green {
//   border-left: $borderWidth solid $color;
//   top: 0;
//   bottom: 0;
//   right: 0;
//   width: $greenWidth;
//   background-color: #FFBE5B;
// }
  /* Similar code for div.orange and div.white */