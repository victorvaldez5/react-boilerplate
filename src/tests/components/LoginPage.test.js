import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage'; 
import { start } from 'repl';

let startLogin, wrapper;

beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<LoginPage startLogin={startLogin} />);
})
test('Should correctly render login page', () => {
  expect(wrapper).toMatchSnapshot();
});

test("Should call startLogin on button click", () => {
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenLastCalledWith();
});