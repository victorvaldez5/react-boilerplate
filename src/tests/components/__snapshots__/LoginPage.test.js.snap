// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      tagline for app!
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
